### ISSUE TITLE
PDFObject does not allow apostrophe in URL

### ISSUE THREAD

**Issue by jackkinsella** [pipwerks/PDFObject#173]
If url is something like "occupier's liability.pdf", then PDFObject fails. This is because of how the embed element is currently created.

```javascript
// generateEmbedElement()

targetNode.innerHTML = "<embed " + id + " class='pdfobject' src='" + url + pdfOpenFragment + "' type='application/pdf' style='overflow: auto; " + style + "'/>";
```

**Comment by jackkinsella** [comment]
The issue doesn't seem to occur if the object is built up as follows:

```javascript
const embed = document.createElement('embed')
embed.src = url
```

**Comment by pipwerks** [comment]
HI Jack

Thanks for submitting. My first instinct is to say that anyone attempting to use an apostrophe in a URL should know to escape it first, as this is a common issue in JS. 

However I do like your suggested workaround; I will look into it.

Thanks

**Comment by jackkinsella** [comment]
You're right about 'buyer beware', but then again, a small change could add a handy safety latch to help people out from latent bugs — esp those dealing with large clumps of dirty legacy code where it's unclear whether the rules about escaping are properly enforced.

**Comment by jackkinsella** [comment]
happy to do a PR if you're interested. LMK

**Comment by PabloAH** [comment]
A smaller change, just to avoid that string function, is to escape the ' 
Assuming you have escaped all characters but ' which is a valid filename character

        targetNode.innerHTML = "<embed " + id + " class='pdfobject' src='" + url.replace("'", "&#39;") + pdfOpenFragment + "' type='application/pdf' style='overflow: auto; " + style + "'/>";

**Comment by pipwerks** [comment]
This has been addressed in PDFObject 2.2.0, which is in testing and will be released shortly

**Comment by pipwerks** [comment]
2.2.0 tests are available here if you have a moment to try them: https://pdfobject.com/dev/
Thanks
